_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1069,8 +1069,7 @@ private function _process_onboarding( $data ) {
 
 		// Setup contact page and add a form and/or business info
 		$contact_page = '';
-
-		if ( isset( $data['addContactForm'] ) && $data['addContactForm'] ) {
+		if ( ! empty( $data['addContactForm'] ) && ! get_option( 'jpo_contact_page' ) ) {
```

Comment:
Note that this won't work in an ideal way if we've deleted the corresponding page. In that particular case, the option will still be there, containing a page ID, but the page with that ID will not be there; and in that case we'd want to create a new page with a form shortcode and update the option.

So, do we want to get this a step further and check if the page with that ID actually exists? Which would be something like `get_post( get_option( 'jpo_contact_page' ) ) === null`? And insert the page with the form only if the page doesn't exist? I'd say this is a worthy improvement.

Which of course naturally brings a follow up question: do we want to get it even further by checking whether that page contains the contact form shortcode? My $0.02: I'd say "No" to this last question, because that case would imply that the user has modified the page content intentionally, so we wouldn't want to interfere. 

What do you think?

Reply:
🕵️ 💯